src/components/home_components/AllArticles.jsx

```diff
@@ -1,25 +1,27 @@
 import {useEffect, useState} from "react";
-import {Link} from "react-router-dom";
+import {useLocation} from "react-router-dom";
 
 import Loader from "../Loader.jsx";
 import {getArticles} from "../../api.jsx";
 
 function AllArticles(){
     const [articles, setArticles] = useState([])
     const [isLoading, setLoading] = useState(true)
+    let location = useLocation()
+    let query = new URLSearchParams(location.search).get('topic')
 
 
 
 
 
 
     useEffect(() => {
-        getArticles().then(articlesCards=>{
+        getArticles(query).then(articlesCards=>{
         setArticles(articlesCards)
         setLoading(false)
         })
 
-    }, []);
+    }, [query]);
```

Comment:
You basically have the mechanics set up to do this as I suggested up at the top with the middle step of the topic page removed. You will just need to adjust your logic to make sure users can select a topic so it's able to change